File: concrete/src/test/java/pcb/uwu/sandbox/MitchGoesRunning.java
package pcb.uwu.sandbox;

import org.junit.Test;
import pcb.uwu.amount.base.Length;
import pcb.uwu.amount.base.Time;
import pcb.uwu.amount.derived.mechanics.Speed;

import static java.math.MathContext.DECIMAL64;
import static org.junit.Assert.assertEquals;
import static pcb.uwu.unit.base.MeterUnit.METER;
import static pcb.uwu.unit.base.MileUnit.MILE;
import static pcb.uwu.unit.base.MinuteUnit.MINUTE;
import static pcb.uwu.unit.derived.mechanics.MeterPerSecondUnit.METER_PER_SECOND;

public class MitchGoesRunning {

	@Test
	public void testMitchGoesRunning() {
		Length mitchDistance = new Length(1000, METER);
		Time mitchTime = new Time(6, MINUTE);
		Speed mitchSpeed = mitchDistance.dividedBy(mitchTime, DECIMAL64);

		Length wifeDistance = new Length(1, MILE);
		Time wifeTime = new Time(11, MINUTE);
		Speed wifeSpeed = wifeDistance.dividedBy(wifeTime, DECIMAL64);

		assertEquals("2.777777777777778 s⁻¹⋅m", mitchSpeed.in(METER_PER_SECOND).toString());
		assertEquals("2.438400000000000 s⁻¹⋅m", wifeSpeed.in(METER_PER_SECOND).toString());
	}
}

File: concrete/src/test/java/pcb/uwu/sandbox/MitchGoesRunning.kt
package pcb.uwu.sandbox

import org.junit.Assert
import org.junit.Test
import pcb.uwu.amount.base.Length
import pcb.uwu.amount.base.Time
import pcb.uwu.unit.base.MeterUnit.METER
import pcb.uwu.unit.base.MileUnit
import pcb.uwu.unit.base.MinuteUnit.MINUTE
import pcb.uwu.unit.derived.mechanics.MeterPerSecondUnit.METER_PER_SECOND

class MitchGoesRunning
{
    @Test
    fun testMitchGoesRunning()
    {
        val mitchDistance = Length(amount = 1000, unit = METER)
        val mitchTime = Time(amount = 6, unit = MINUTE)
        val mitchSpeed = mitchDistance.div(mitchTime)
        val wifeDistance = Length(amount = 1, unit = MileUnit.MILE)
        val wifeTime = Time(amount = 11, unit = MINUTE)
        val wifeSpeed = wifeDistance.div(wifeTime)
        Assert.assertEquals("2.777777777777777777777777777777778 m⋅s⁻¹", (mitchSpeed to METER_PER_SECOND).toString())
        Assert.assertEquals("2.4384 m⋅s⁻¹", (wifeSpeed to METER_PER_SECOND).toString())
    }
}